File: app/src/main/java/com/anokmik/tripassistant/author/AuthorPresenter.java
package com.anokmik.tripassistant.author;

import com.anokmik.persistence.model.Author;
import com.anokmik.persistence.repository.AuthorRepository;

public final class AuthorPresenter implements AuthorContract.Presenter {

    private final AuthorContract.View view;
    private final AuthorRepository authorRepository;

    public AuthorPresenter(AuthorContract.View view) {
        this.view = view;
        this.authorRepository = new AuthorRepository();
    }

    @Override
    public void validateAuthor() {
        Author author = authorRepository.getActive();
        if (author != null) {
            view.showProfile(author);
        } else {
            view.showLogin();
        }
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/author/AuthorPresenter.kt
package com.anokmik.tripassistant.author

import com.anokmik.persistence.repository.AuthorRepository

class AuthorPresenter(private val view: AuthorContract.View) : AuthorContract.Presenter {

    private val authorRepository = AuthorRepository()

    override fun validateAuthor() {
        val author = authorRepository.getActive()
        if (author != null) view.showProfile(author) else view.showLogin()
    }

}